### PR TITLE
Add user services, repository and unit tests. 

### PR DESCRIPTION
**Pull request by AliUnipal**
- Cleanup the chat services and repository by seperating the user repo logic.
- Add in memory user repo implementation.

### CODE REVIEW

File: internal/service/chatsvc/repo/inmemchatrepo/repo.go [hunk 1]
```diff
@@ -4,55 +4,56 @@ import (
...
-	if !ok {
-		return errors.New("user one does not exist")
-	}
-	u2, ok := r.users[in.UserTwoID]
-	if !ok {
-		return errors.New("user two does not exist")
-	}
-	ids := []string{in.UserOneID.String(), in.UserTwoID.String()}
+type userRepository interface {
+	GetUser(ctx context.Context, id uuid.UUID) (userRepo.CreateUserInput, error)
```
Comment (almahoozi) [inline internal/service/chatsvc/repo/inmemchatrepo/repo.go:28]: Yep. Continuing on the first comment: we should have our own type, and use a wrapped version of the user repository. We can discuss this if unclear.

File: internal/service/chatsvc/service.go [hunk 2]
```diff
@@ -2,11 +2,10 @@ package chatsvc
 
 import (
 	"context"
-	"errors"
 	"github.com/AliUnipal/chat/internal/models/chat"
 	"github.com/AliUnipal/chat/internal/models/message"
 	"github.com/AliUnipal/chat/internal/models/user"
-	"github.com/AliUnipal/chat/internal/service/chatsvc/repo"
+	chatRepo "github.com/AliUnipal/chat/internal/service/chatsvc/repo"
```
Comment (almahoozi) [inline internal/service/chatsvc/service.go:8]: Why is this aliased? There's only one repo here.

File: internal/service/chatsvc/repo/inmemchatrepo/repo.go (hunk 1 above)
Comment (AliUnipal) [reply internal/service/chatsvc/repo/inmemchatrepo/repo.go:28]: Let's discuss this

File: internal/service/usersvc/service.go [hunk 3]
```diff
@@ -42,6 +43,10 @@ func (s *service) CreateUser(ctx context.Context, in CreateUserInput) (uuid.UUID
 	if in.Username == "" {
 		return uuid.Nil, errors.New("username is required")
 	}
+	u, err := url.ParseRequestURI(in.ImageURL)
```
Comment (almahoozi) [inline internal/service/usersvc/service.go:46]: You should only check the URL if it's not empty - since it's optional I expect this will fail if you don't provide a URL.
You also don't need to check the values; if it's an error it's invalid, otherwise it's ok.

File: internal/service/usersvc/service_test.go [hunk 4]
```diff
@@ -73,10 +73,27 @@ func TestCreateUser_ReturnErrorOnEmptyUsername(t *testing.T) {
 	}
 }
 
+func TestCreateUser_ReturnErrorOnInvalidImageURL(t *testing.T) {
```
Comment (almahoozi) [inline internal/service/usersvc/service_test.go:76]: Please add one more test which has empty URL; validate that that works correctly

File: internal/service/usersvc/service_test.go [hunk 5]
```diff
@@ -73,10 +73,27 @@ func TestCreateUser_ReturnErrorOnEmptyUsername(t *testing.T) {
...
+	ctx := context.Background()
```
Comment (almahoozi) [inline internal/service/usersvc/service_test.go:77]: In tests it's better to use `t.Context()`
